### PR TITLE
Ensure the server and client are valid when sending / receiving hot reload messages

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
I'm going to get this in rc1, give it some time to bake.

File: src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js [hunk 1]
```diff
@@ -187,8 +192,38 @@ setTimeout(async function () {
...
+      }
+      return buf;
+    }
+  }
+
   function getWebSocket(url) {
     return new Promise((resolve, reject) => {
+      const protocol = sharedSecret && sharedSecret.encryptedSharedSecret;
+      if (protocol) {
+        url += `?p=${encodeURIComponent(protocol)}`;
```
Comment (BrennanConroy) [inline src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js:225]: This is a query string not a websocket subprotocol

File: src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js [hunk 2]
```diff
@@ -187,9 +194,37 @@ setTimeout(async function () {
...
+      return buf;
+    }
+  }
+
   function getWebSocket(url) {
     return new Promise((resolve, reject) => {
-      const webSocket = new WebSocket(url);
+      const encryptedSecret = sharedSecret && sharedSecret.encryptedSharedSecret;
+      const protocol = encryptedSecret ? encodeURIComponent(encryptedSecret) : null;
+      const webSocket = new WebSocket(url, protocol);
```
Comment (pranavkm) [inline src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js:227]: @BrennanConroy updated.

Comment (BrennanConroy) [reply src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js:227]: Are there end-to-end tests? Looks like nothing failed with the query string

Comment (pranavkm) [reply src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js:227]: It's going to be tricky to add browser based automated tests to this repo. We don't have a very good track record of making Selenium / Playwrite tests reliable and I wouldn't want them to destabilize the SDK repo. For the interim, my plan is to add some automated tests around the watch inner loop and see where that takes us.

**Pull request merged**